src/runtime/GlobalObjectBuiltinString.cpp

```diff
@@ -307,14 +307,60 @@ static Value builtinStringRepeat(ExecutionState& state, Value thisValue, size_t
 
 static Value builtinStringReplace(ExecutionState& state, Value thisValue, size_t argc, Value* argv, bool isNewExpression)
 {
+    ASSERT(argc == 0 || argv != nullptr);
+#ifdef ESCARGOT_ENABLE_ES2015
+    RESOLVE_THIS_BINDING_TO_OBJECT(obj, Object, replace);
```

Comment:
`RESOLVE_THIS_BINDING_TO_OBJECT` ? The standard says `Let O be RequireObjectCoercible(this value).`. These are not compatible.